x/profiles/keeper/keeper_relationships.go

```diff
@@ -50,24 +59,10 @@ func (k Keeper) GetAllRelationships(ctx sdk.Context) []types.Relationship {
 // RemoveRelationship allows to delete the relationship between the given user and his counterparty
 func (k Keeper) RemoveRelationship(ctx sdk.Context, relationship types.Relationship) error {
 	store := ctx.KVStore(k.storeKey)
-	key := types.RelationshipsStoreKey(relationship.Creator)
-
-	relationships := types.MustUnmarshalRelationships(k.cdc, store.Get(key))
-	relationships, found := types.RemoveRelationship(relationships, relationship)
-
-	// The relationship didn't exist, so return an error
-	if !found {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest,
-			"relationship between %s and %s for subspace %s not found",
-			relationship.Creator, relationship.Recipient, relationship.Subspace)
-	}
-
-	// Delete the key if no relationships are left.
-	// This cleans up the store avoiding export/import tests to fail due to a different number of keys present.
-	if len(relationships) == 0 {
-		store.Delete(key)
-	} else {
-		store.Set(key, types.MustMarshalRelationships(k.cdc, relationships))
+	key := types.RelationshipsStoreKey(relationship.Creator, relationship.Subspace, relationship.Recipient)
+	if !store.Has(key) {
+		return fmt.Errorf("relationship does not exist")
```

Comment:
I think we should stick with the old error: 

```suggestion
		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest,
			"relationship between %s and %s for subspace %s not found",
			relationship.Creator, relationship.Recipient, relationship.Subspace)
```